src/core/mac/data_poll_sender.cpp

```diff
@@ -496,8 +536,88 @@ void DataPollSender::HandlePollTimer(Timer &aTimer)
 
 uint32_t DataPollSender::GetDefaultPollPeriod(void) const
 {
-    return Time::SecToMsec(Get<Mle::MleRouter>().GetTimeout()) -
-           static_cast<uint32_t>(kRetxPollPeriod) * kMaxPollRetxAttempts;
+    uint32_t period = Time::SecToMsec(Get<Mle::MleRouter>().GetTimeout());
+
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+    if (Get<Mac::Mac>().IsCslEnabled())
+    {
+        period = OT_MIN(period, Time::SecToMsec(Get<Mac::Mac>().GetCslTimeout()));
+    }
+#endif
+
+    period -= (static_cast<uint32_t>(kRetxPollPeriod) * kMaxPollRetxAttempts);
+
+    return period;
+}
+
+Mac::TxFrame *DataPollSender::PrepareDataRequest(Mac::TxFrames &aTxFrames)
+{
+    Mac::TxFrame *frame = nullptr;
+    Mac::Address  src, dst;
+    uint16_t      fcf;
+    bool          iePresent;
+
+#if OPENTHREAD_CONFIG_MULTI_RADIO
+    Mac::RadioType radio;
+
+    SuccessOrExit(GetPollDestinationAddress(dst, radio));
+    frame = &aTxFrames.GetTxFrame(radio);
+#else
+    SuccessOrExit(GetPollDestinationAddress(dst));
+    frame = &aTxFrames.GetTxFrame();
+#endif
+
+    fcf = Mac::Frame::kFcfFrameMacCmd | Mac::Frame::kFcfPanidCompression | Mac::Frame::kFcfAckRequest |
+          Mac::Frame::kFcfSecurityEnabled;
+
+    iePresent = Get<MeshForwarder>().CalcIePresent(nullptr);
+
+    if (iePresent)
+    {
+        fcf |= Mac::Frame::kFcfIePresent;
+    }
+
+    fcf |= Get<MeshForwarder>().CalcFrameVersion(Get<NeighborTable>().FindNeighbor(dst), iePresent);
+
+    if (dst.IsExtended())
+    {
+        fcf |= Mac::Frame::kFcfDstAddrExt | Mac::Frame::kFcfSrcAddrExt;
+        src.SetExtended(Get<Mac::Mac>().GetExtAddress());
+    }
+    else
+    {
+        fcf |= Mac::Frame::kFcfDstAddrShort | Mac::Frame::kFcfSrcAddrShort;
+        src.SetShort(Get<Mac::Mac>().GetShortAddress());
+    }
+
+    frame->InitMacHeader(fcf, Mac::Frame::kKeyIdMode1 | Mac::Frame::kSecEncMic32);
+
+    if (frame->IsDstPanIdPresent())
+    {
+        frame->SetDstPanId(Get<Mac::Mac>().GetPanId());
+    }
+
+    frame->SetSrcAddr(src);
+    frame->SetDstAddr(dst);
+#if OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT
+    if (iePresent)
+    {
+        Get<MeshForwarder>().AppendHeaderIe(nullptr, *frame);
+    }
+
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+    if (frame->GetHeaderIe(Mac::CslIe::kHeaderIeId) != nullptr)
+    {
+        // Disable frame retransmission when the data poll has CSL IE included
+        aTxFrames.SetMaxFrameRetries(0);
+    }
```

Comment:
It makes sense to move this method here in `DataPollSender` from `Mac`. 

Just double-chekcing that the `if` block above is the only addition to this method (compared to what it was in `Mac`). Is this correct? Thanks.

Reply:
That's right.